crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -230,19 +231,24 @@ where
                     let context = self.semantic_model.execution_context();
                     let exceptions = self.semantic_model.exceptions();
                     let scope = &mut self.semantic_model.scopes[self.semantic_model.scope_id];
-                    let usage = Some((self.semantic_model.scope_id, stmt.range()));
                     for (name, range) in names.iter().zip(ranges.iter()) {
-                        let id = self.semantic_model.bindings.push(Binding {
+                        // Add a reference to the global binding.
+                        let reference_id = self.semantic_model.references.push(
```

Comment:
I don't really have intuition on whether we should be defining methods like `push_reference` on `SemanticModel`, or on `References` and accessing them via attribute like this.

Reply:
Creating a method on the semantic model makes the call shorter and allows to default to the current scope (so that you don't need to specify current scope everywhere)

I generally prefer to encapsulate all mutations, to have better control over how the data structures are modified (and only allow the operations that are valid)